### PR TITLE
add some doctests to Dates

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: base/dates/accessors.jl [hunk 1]
```diff
@@ -85,8 +85,21 @@ end
 Return the [ISO week date](https://en.wikipedia.org/wiki/ISO_week_date) of a `Date` or
 `DateTime` as an [`Int64`](@ref). Note that the first week of a year is the week that
 contains the first Thursday of the year which can result in dates prior to January 4th
-being in the last week of the previous year. For example `week(Date(2005,1,1))` is the 53rd
+being in the last week of the previous year. For example `week(Date(2005, 1, 1))` is the 53rd
```
Comment (kshyatt) [inline base/dates/accessors.jl]: "For example," if we're already changing this line :)

File: base/dates/accessors.jl [hunk 2]
```diff
@@ -85,8 +85,21 @@ end
 Return the [ISO week date](https://en.wikipedia.org/wiki/ISO_week_date) of a `Date` or
 `DateTime` as an [`Int64`](@ref). Note that the first week of a year is the week that
 contains the first Thursday of the year which can result in dates prior to January 4th
```
Comment (kshyatt) [inline base/dates/accessors.jl]: I think we want a comma before "which" as well

File: base/dates/adjusters.jl [hunk 3]
```diff
@@ -113,6 +163,15 @@ firstdayofquarter(dt::DateTime) = DateTime(firstdayofquarter(Date(dt)))
     lastdayofquarter(dt::TimeType) -> TimeType
 
 Adjusts `dt` to the last day of its quarter.
+
+# Example
```
Comment (kshyatt) [inline base/dates/adjusters.jl]: Examples, since you have 2, right?

File: base/dates/query.jl [hunk 4]
```diff
@@ -90,6 +90,15 @@ dayofweek(days) = mod1(days, 7)
     daysinyear(dt::TimeType) -> Int
 
 Returns 366 if the year of `dt` is a leap year, otherwise returns 365.
+
+# Example
```
Comment (kshyatt) [inline base/dates/query.jl]: Examples again

File: base/dates/query.jl [hunk 5]
```diff
@@ -149,6 +175,18 @@ issunday(dt::TimeType) = dayofweek(dt) == Sun
 For the day of week of `dt`, returns which number it is in `dt`'s month. So if the day of
 the week of `dt` is Monday, then `1 = First Monday of the month, 2 = Second Monday of the
 month, etc.` In the range 1:5.
+
+# Example
```
Comment (kshyatt) [inline base/dates/query.jl]: Examples

File: base/dates/query.jl [hunk 6]
```diff
@@ -168,6 +206,15 @@ For the day of week of `dt`, returns the total number of that day of the week in
 month. Returns 4 or 5. Useful in temporal expressions for specifying the last day of a week
 in a month by including `dayofweekofmonth(dt) == daysofweekinmonth(dt)` in the adjuster
 function.
+
+# Example
```
Comment (kshyatt) [inline base/dates/query.jl]: Examples

File: base/dates/query.jl [hunk 7]
```diff
@@ -200,6 +253,11 @@ end
     monthabbr(dt::TimeType; locale="english") -> AbstractString
 
 Return the abbreviated month name of the `Date` or `DateTime` in the given `locale`.
+
```
Comment (kshyatt) [inline base/dates/query.jl]: Example

File: base/dates/query.jl [hunk 8]
```diff
@@ -217,6 +287,15 @@ daysinmonth(dt::TimeType) = ((y, m) = yearmonth(dt); return daysinmonth(y, m))
     isleapyear(dt::TimeType) -> Bool
 
 Returns `true` if the year of `dt` is a leap year.
+
+# Example
```
Comment (kshyatt) [inline base/dates/query.jl]: Examples

**Review by kshyatt — APPROVED** [review]
LGTM modulo my extremely nit-picky changes and your fixing the whitespace

File: base/dates/accessors.jl [hunk 9]
```diff
@@ -85,8 +85,21 @@ end
...
+julia> Dates.week(Date(1989, 6, 22))
+25
+
+julia> Dates.week(Date(2005, 1, 1))
+53
+
+julia> Dates.week(Date(2004, 12, 31))
+53
+```
+
```
Comment (tkelman) [inline base/dates/accessors.jl]: no empty line here

File: base/dates/query.jl [hunk 10]
```diff
@@ -149,6 +175,18 @@ issunday(dt::TimeType) = dayofweek(dt) == Sun
 For the day of week of `dt`, returns which number it is in `dt`'s month. So if the day of
 the week of `dt` is Monday, then `1 = First Monday of the month, 2 = Second Monday of the
 month, etc.` In the range 1:5.
+
+# Example
+```jldoctest
+julia> Dates.dayofweekofmonth(Date("2000-02-1"))
```
Comment (tkelman) [inline base/dates/query.jl]: are these intentionally using 1 and 8 instead of 01 and 08?

**Comment by KristofferC** [comment]
Updated from review. FWIW I think we should just have plural "#Examples" all the time.

**Review by Sacha0 — APPROVED** [review]
superficially lgtm! :)

**Pull request merged**